### ISSUE TITLE
Regression `@tanstack/start`: generated d.ts files for ESM don’t have file extensions

### ISSUE THREAD

**Issue by lo1tuma** [TanStack/router#2223]
### Describe the bug

When having `moduleResolution` set to `node16` typescript complains about imports without file extension.

```
node_modules/@tanstack/start/dist/esm/client/index.d.ts:8:70 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '../constants.js'?

8 export { serverFnPayloadTypeHeader, serverFnReturnTypeHeader, } from '../constants';
                                                                       ~~~~~~~~~~~~~~


Found 1 error in node_modules/@tanstack/start/dist/esm/client/index.d.ts:8
```

### Your Example Website or App

https://arethetypeswrong.github.io/?p=%40tanstack%2Fstart%401.51.4

### Steps to Reproduce the Bug or Issue

1. set `moduleResolution` in tsconfig to `node16`
1. import something from `@tanstack/start`
1. run the typescript compiler

### Expected behavior

I would expect the file extensions in the .d.ts file to be present.

### Platform

- Version: 1.51.4

**Comment by schiller-manuel** [comment]
this is now corrected in v1.51.6 https://arethetypeswrong.github.io/?p=%40tanstack%2Fstart%401.51.6

**Comment by lo1tuma** [comment]
Thank you!
